### DISCUSSION TITLE
When will LMDB Store no longer be marked experimental?

### DISCUSSION THREAD

**Discussion by daltontc** [eclipse-rdf4j/rdf4j#4786] [General]
Hello, I was curious when the LMDB store will no longer be experimental and be production ready? Is that on the roadmap for 5.0?

Thanks,
Tom

**Comment by kenwenzel** [comment]
Hi Tom,

I'm thinking about changing the encoding of literals:
https://github.com/eclipse-rdf4j/rdf4j/issues/4774

This would require a change of the database structure and would not be backwards compatible. But maybe this is also something that could be done after releasing 5.0.

@hmottestad @abrokenjester What are the plans for 5.0?

Best regards,
Ken

**Reply by hmottestad** [reply]
You can remove the experimental status in a major or minor release, so you can wait for 5.1 or any later minor release. 

Before the 5.0 release we should probably go through your packages and make sure that things are marked as internal use. Even if you move from experimental to stable you might not want all the internal interfaces and public methods to be extendable by others. 

Once stable though we take great care to make sure that changes to on-disk data structures are migrated automatically with no user interaction. It's a nice thing to do even if it's marked as experimental, but I would wait for the last big changes to the internal data structures before calling it stable.
